File: makeGifMaker.pl
use strict;
use warnings;
use utf8;
use Encode;

my $step = 0;
my $gene = 0;
my ($fstats) = @ARGV;
my @column;
my $dir;

die 'no text' unless defined $fstats;
$fstats = decode('utf-8', $fstats);
open(my $ifh, "<", $fstats)
  or die "Cannnot open $fstats: $!";
while( my $line = readline($ifh) ){
  chomp($line);
  @column = split(/\t/, $line);
  if( $column[0] eq 'gene'){
     $gene = int($column[1]);
  }
}
close $ifh;	

$dir = sprintf("./stateLogs/%08d", $gene);
printf("#!/bin/sh\n");
printf("/usr/local/bin/convert -delay 10 %s/*.svg /home/ikatake/www/wetsteam/lifegamebot/gifs/%08d.gif\n",$dir ,$gene);
#	printf("rm ${dir}/*.svg\n");	

File: makeGifMaker.py
import sys
import common

### ここから本体 ###
#ループ情報を読み取るファイル
file_name:str = sys.argv[1]
step:int = 0
gene:int = 0
loop_from:int = 0
#print('[DBGmakeGifMaker.py]' + file_name)

#ループが確定したgeneration・step・ループ元stepを読み取る。
gene, step, loop_from = common.readLoopFile(file_name)

in_arg_pngs:str = './pngs/' + '{:08}/*.png'.format(gene)
out_arg_gif:str = '/home/ikatake/www/wetsteam/lifegamebot/gifs/'
out_arg_gif += '{:08}.gif'.format(gene)
#gene_folder = common.state_log_dir+ '{:08}/'.format(gene)
print('#!/bin/sh')
print('/usr/local/bin/convert -delay 10 ' + in_arg_pngs + ' ' + out_arg_gif)
